src/serverModel.ts

```diff
@@ -153,28 +304,37 @@ export default function ServerModel(mpInstance) {
             };
         }
 
-        return jsonObject;
+        return jsonObject as IConsentStateDTO;
     };
 
-    this.createEventObject = function(event, user) {
-        var uploadObject = {};
-        var eventObject = {};
+    this.createEventObject = function(
+        event: BaseEvent,
+        user?: MParticleUser
+    ): SDKEvent | IUploadObject {
+        var uploadObject: Partial<IUploadObject> = {};
+        var eventObject: Partial<SDKEvent> = {};
 
+        // TODO: What is the purpose of setting OptOut to the inverse of the Store's
+        //       isEnabled property and not just `false`?
```

Comment:
There is a method, [mParticle.setOptOut(boolean)](https://github.com/mParticle/mparticle-web-sdk/blob/master/src/mp-instance.js#L1087-L1116), which accepts a boolean and that sets the `isEnabled` flag to be used elsewhere in the SDK for if it should log or not (line 321 below).  It doesn't set opt out to true all the time, which isn't clear from the code here.